src/main/java/io/tiledb/java/api/Array.java

```diff
@@ -456,46 +456,36 @@ public static void create(
    */
   public HashMap<String, Pair> nonEmptyDomain() throws TileDBError {
     checkIsOpen();
-    HashMap<String, Pair> ret = new HashMap<String, Pair>();
-    try (Domain domain = schema.getDomain();
-        NativeArray domainArray =
-            new NativeArray(ctx, 2 * (int) domain.getRank(), domain.getType())) {
-      SWIGTYPE_p_int emptyp = tiledb.new_intp();
-      try {
-        ctx.handleError(
-            tiledb.tiledb_array_get_non_empty_domain(
-                ctx.getCtxp(), arrayp, domainArray.toVoidPointer(), emptyp));
-        if (tiledb.intp_value(emptyp) == 1) {
-          return ret;
-        }
-      } finally {
-        tiledb.delete_intp(emptyp);
-      }
-      for (int i = 0; i < domain.getRank(); i++) {
-        try (Dimension d = domain.getDimension(i)) {
-          ret.put(
-              d.getName(),
-              new Pair(domainArray.getItem((2 * i) + 0), domainArray.getItem((2 * i) + 1)));
-        }
+    HashMap<String, Pair> ret = new HashMap<>();
+    try (Domain domain = schema.getDomain()) {
+      long numDims = domain.getNDim();
+      for (long dimIdx = 0; dimIdx < numDims; ++dimIdx) {
+        Dimension dimension = domain.getDimension(dimIdx);
```

Comment:
Does the dimension need to be inside a try for the autoclosure?

Reply:
In fact no, this is a leftover from the old code. Fixed that.